### PR TITLE
Ensure cadence don't access unexpected operations while parsing (part 3)

### PR DESCRIPTION
**Pull request by pattyshack**

### CODE REVIEW

File: fvm/environment/parse_restricted_checker.go [hunk 1]
```diff
@@ -133,3 +167,54 @@ func parseRestrict4Arg1Ret[
...
+) {
+	err := parseRestricted(txnState, opCode)
+	if err != nil {
+		var value1 Ret1T
+		var value2 Ret2T
+		return value1, value2, err
+	}
+
+	return callback(arg)
+}
```
Comment (Tonix517) [inline fvm/environment/parse_restricted_checker.go:220]: iirc golang does not have plan to support generic variadic functions. i'm thinking combining https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source and "text/template" may save us some typing here. but just for entertainment purposes, it is too hacky for production.

**Review by Tonix517 — APPROVED** [review]

File: fvm/environment/parse_restricted_checker.go (hunk 1 above)
Comment (pattyshack) [reply fvm/environment/parse_restricted_checker.go:220]: not supporting adhoc (i.e., overloading / variadic) polymorphism is the correct thing to do from the language designer's perspective.  unlike parametric polymorphism, there's no solid theoretical foundation for adhoc polymorphism.

Comment (Tonix517) [reply fvm/environment/parse_restricted_checker.go:220]: this is why i hate C++ ;)

**Review by SaveTheRbtz — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Review by janezpodhostnik — APPROVED** [review]
Hmmm... It might be time to unify `opCode`, `SpanName` and `common.ComputationKind` and use a operation wrapper like the `parseRestricted` to not only check for `IsParseRestricted` but also do:
```
defer info.tracer.StartSpanFromRoot(opCode).End()
err := info.meter.MeterComputation(opCode, intensity())
```

**Comment by pattyshack** [comment]
yeah, that's reasonable.

**Comment by Tonix517** [comment]
+1
> Hmmm... It might be time to unify `opCode`, `SpanName` and `common.ComputationKind` and use a operation wrapper like the `parseRestricted` to not only check for `IsParseRestricted` but also do:
> 
> ```
> defer info.tracer.StartSpanFromRoot(opCode).End()
> err := info.meter.MeterComputation(opCode, intensity())
> ```

**Pull request merged**